src/django/api/models.py

```diff
@@ -2314,6 +2322,73 @@ class ExtendedField(models.Model):
     history = HistoricalRecords()
 
 
+@transaction.atomic
+def index_custom_text(facility_ids=list):
+    # If passed an empty array, update all facilities (where applicable)
+    if len(facility_ids) == 0:
+        print('Indexing custom text for all facilities...')
+        facility_ids = Facility.objects.all().values_list('id', flat=True)
+
+    # Get a list of searchable embed fields
+    fields_filter = (Q(searchable=True) & Q(visible=True)
+                     & Q(embed_config__isnull=False))
+    fields = EmbedField.objects.filter(fields_filter)
+
+    # Get a list of contributors with searchable fields
+    contributor_ids = fields.values_list('embed_config__contributor',
+                                         flat=True).distinct()
+
+    # Get a list of active FacilityListItems for the given facilities.
+    # Only include list items where the contributors have searchable fields.
+    # Select the most recent item for each facility for each contributor.
+    items_filter = (Q(facility_id__in=facility_ids)
+                    & Q(source__contributor_id__in=contributor_ids)
+                    & Q(source__is_active=True)
+                    & Q(facilitymatch__is_active=True))
+    items = FacilityListItem.objects.filter(items_filter) \
+        .distinct('facility__id', 'source__contributor__id') \
+        .order_by('facility__id', 'source__contributor__id', '-created_at') \
+        .iterator()
+
+    custom_fields = defaultdict(str)
+    contributor_fields = defaultdict(list)
+
+    for item in items:
+        contributor_id = item.source.contributor.id
+
+        # Calculate a list of searchable fields for the item's contributor
+        if len(contributor_fields[contributor_id]) == 0:
+            contributor_fields[contributor_id] = fields.filter(
+                    Q(embed_config__contributor__id=contributor_id)) \
+                    .values_list('column_name', flat=True)
+
+        formatted_fields = [{'value': '', 'column_name': f} for f
```

Comment:
We have to reformat the fields for each facility instead of using the dictionary to prevent getting values from other facilities belonging to that contributor mixed in, but the dictionary still prevents requerying for the contributor fields.